espresso/environment/8_reorg_test.go

```diff
@@ -0,0 +1,93 @@
+package environment_test
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	env "github.com/ethereum-optimism/optimism/espresso/environment"
+	"github.com/ethereum-optimism/optimism/op-e2e/system/e2esys"
+	"github.com/stretchr/testify/require"
+)
+
+// TestBatcherWaitForFinality is a test that attempts to make sure that the batcher waits for the
+// derived L1 block to be finalized before submitting a new block.
+//
+// This tests is designed to evaluate Test 8.2 as outlined within the Espresso Celo Integration
+// plan. It has stated task definition as follows:
+//
+//	Arrange:
+//		Run the sequencer and the batcher in Espresso mode.
+//	Act:
+//		Wait until a new block is finalized.
+//	Assert:
+//		The batcher doesn't submit a block without finalized L1 origin to the L1.
+//		After the L1 origin is finalized, the batcher submits the block.
+func TestBatcherWaitForFinality(t *testing.T) {
+	// Basic test setup.
+	ctx, cancel := context.WithTimeout(context.Background(), 2 * time.Minute)
+	defer cancel()
+	launcher := new(env.EspressoDevNodeLauncherDocker)
+
+	// Set NonFinalizedProposals to true and SequencerUseFinalized to false, to make sure we are
+	// testing how the batcher handles the finality.
+	system, espressoDevNode, err := launcher.StartDevNet(ctx, t, env.WithL1FinalizedDistance(0), env.WithNonFinalizedProposals(true), env.WithSequencerUseFinalized(false))
```

Comment:
I believe that the use finalized flag on the sequencer is set to `false` by default, so this call with `env.WithSequencerUseFinalized` is unncessary.

Reply:
Yeah it's unnecessary for now. I just wanted to set it explicitly in case in the future we'll make it `true` by default, which we'll likely do.